packages/teams-js/src/public/media.ts

```diff
@@ -412,30 +414,42 @@ export namespace media {
      * @param mediaEvent indicates what the event that needs to be signaled to the host client
      * Optional; @param callback is used to send app if host client has successfully handled the notification event or not
      */
-    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): void {
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): Promise<void> {
       ensureInitialized(FrameContexts.content, FrameContexts.task);
-      const err = isApiSupportedOnMobile(nonFullScreenVideoModeAPISupportVersion);
-      if (err) {
-        if (callback) {
-          callback(err);
-        }
-        return;
-      }
+
+      const promise = Promise.resolve().then(() =>
+        throwExceptionIfMobileApiIsNotSupported(nonFullScreenVideoModeAPISupportVersion),
+      );
 
       const params: MediaControllerParam = { mediaType: this.getMediaType(), mediaControllerEvent: mediaEvent };
+
       sendMessageToParent('media.controller', [params], (err?: SdkError) => {
         if (callback) {
-          callback(err);
+          return callback(err);
+        } else if (err) {
+          return err;
         }
       });
+      return promise;
     }
 
     /**
+     * Function to programatically stop the ongoing media event
+     *
+     * @returns A resolved promise
+     * */
+    public stop(): Promise<void>;
+    /**
+     * @deprecated
+     * As of 2.0.0-beta.3, please use {@link MediaController.stop MediaController.stop(): Promise\<void\>} instead.
+     
      * Function to programatically stop the ongoing media event
      * Optional; @param callback is used to send app if host client has successfully stopped the event or not
      */
-    public stop(callback?: (err?: SdkError) => void): void {
-      this.notifyEventToHost(MediaControllerEvent.StopRecording, callback);
+    public stop(callback: (err?: SdkError) => void): void;
+    public stop(callback?: (err?: SdkError) => void): Promise<void> {
+      const inputFn = (): Promise<void> => this.notifyEventToHost(MediaControllerEvent.StartRecording, callback);
```

Comment:
Curious to hear why you decided not to go with the callCallbackWithErrorOrResultFromPromiseAndReturnPromise  pattern from other back compatible public functions? I don't think what you've done is wrong necessarily, but it does mean that we're not following the same pattern as other similar functions. How come?

Reply:
This is my bad, Jess is doing the right thing here but since inputFn wasn't named the same as the other similar variables I missed it.